autometa/validation/datasets.py

```diff
@@ -0,0 +1,221 @@
+        choices=[
+            "synthetic",
+            "simulated",
+            "all",
+        ],
+        required=True,
+    )
+    parser.add_argument(
+        "--community_size",
+        help="specify a community size to download from",
+        choices=[
+            "78Mbp",
+            "156Mbp",
+            "312Mbp",
+            "625Mbp",
+            "1250Mbp",
+            "2500Mbp",
+            "5000Mbp",
+            "10000Mbp",
+            "all",
+        ],
+        required=True,
+        nargs="+",
+    )
+    parser.add_argument(
+        "--file_names",
+        help="specify a file name to download",
+        choices=[
+            "README.md",
+            "reference_assignments.tsv.gz",
+            "metagenome.fna.gz",
+            "master.tsv.gz",
+            "control_reads.tsv.gz",
+            "control_contigs.tsv.gz",
+            "unclustered_recruitment.tsv.gz",
+            "binning.tsv.gz",
+            "taxonomy.tsv.gz",
+            "lengths.tsv.gz",
+            "coverages.tsv.gz",
+            "gc_content.tsv.gz",
+            "kmers.embedded.tsv.gz",
+            "kmers.tsv.gz",
+            "markers.tsv.gz",
+            "Bacteria.fna.gz",
+            "orfs.faa.gz",
+            "metagenome.filtered.fna.gz",
+            "hmmscan.tsv.gz",
+            "forward_reads.fastq.gz",
+            "reverse_reads.fastq.gz",
+            "all",
+        ],
+        nargs="+",
+        required=True,
+    )
+    parser.add_argument(
+        "--dir_path",
+        help="specify a folder to start the download (several directories will be generated within this folder)",
+        required=True,
+    )
+    args = parser.parse_args()
+
+    community_type = args.community_type
+    if "all" in args.community_size:
+        community_size = (
+            "78Mbp",
+            "156Mbp",
+            "312Mbp",
+            "625Mbp",
+            "1250Mbp",
+            "2500Mbp",
+            "5000Mbp",
+            "10000Mbp",
+        )
+    else:
+        community_size = args.community_size
+    if "all" in args.file_names:
+        file_names = (
+            "README.md",
+            "reference_assignments.tsv.gz",
+            "metagenome.fna.gz",
+            "master.tsv.gz",
+            "control_reads.tsv.gz",
+            "control_contigs.tsv.gz",
+            "unclustered_recruitment.tsv.gz",
+            "binning.tsv.gz",
+            "taxonomy.tsv.gz",
+            "lengths.tsv.gz",
+            "coverages.tsv.gz",
+            "gc_content.tsv.gz",
+            "kmers.embedded.tsv.gz",
+            "kmers.tsv.gz",
+            "markers.tsv.gz",
+            "Bacteria.fna.gz",
+            "orfs.faa.gz",
+            "metagenome.filtered.fna.gz",
+            "hmmscan.tsv.gz",
+            "forward_reads.fastq.gz",
+            "reverse_reads.fastq.gz",
+        )
+    else:
+        file_names = args.file_names
+    dir_path = args.dir_path
+
+    if not internet_is_connected():
```

Comment:
I believe there is an argument for this function that you could pass through argparse. e.g. The ability to ping a different address.

Reply:
I'm thinking it's not worth the extra argparse complexity for including this. If the check fails, the program still tries to run, so I think it's probably best to leave it as a "heads-up" warning.

Reply:
Perhaps. However, how is the program supposed to behave if the internet is not connected? Similarly, adding a parameter to the entrypoint would provide a solution for the edge case Chase mentioned while also helping to account for when the internet is truly not connected, versus just trying to connect to a banned IP address.


```python
## Addition of optional host parameter
parser.add_argumet(
    "--host",
    help="IP address to ping when checking internet connectivity. Note: Will attempt to connect to port 43 on host address (Default is google.com)",
    default="8.8.8.8"
)
...

if not internet_is_connected(host=args.host):
    # Now (assuming the user is aware of their IP bans and has input an appropriate host address) 
    # we know the user is not connected to the internet and can appropriately modify the programs behavior.
    ...

```

Reply:
Ok, I've added that argument.